lib/components/base/ShareModal.vue

```diff
@@ -0,0 +1,219 @@
+<script setup>
+import {
+  Button,
+  Modal,
+  ClipboardCopyIcon,
+  Avatar,
+  LinkIcon,
+  SendIcon,
+  GlobeIcon,
+  TwitterIcon,
+  MastodonIcon,
+  RedditIcon,
+} from '@/components'
+import { computed, ref } from 'vue'
+
+const props = defineProps({
+  header: {
+    type: String,
+    default: 'Share',
+  },
+  shareTitle: {
+    type: String,
+    default: 'Modrinth',
+  },
+  shareText: {
+    type: String,
+    default: 'Share',
+  },
+  link: {
+    type: Boolean,
+    default: false,
+  },
+})
+
+const shareModal = ref(null)
+
+const qrCode = ref(null)
+const qrImage = ref(null)
+const content = ref(null)
+
+const show = async (passedContent) => {
+  content.value = passedContent
+  if (props.link) {
+    const url =
+      'https://api.qrserver.com/v1/create-qr-code/?size=150x150&data=' +
```

Comment:
Surely we don't need to use an external API for this. Quick google search returns https://github.com/cozmo/jsQR